holoviews/core/spaces.py

```diff
@@ -513,6 +515,25 @@ def __call__(self, *args, **kwargs):
         return ret
 
 
+
+class Generator(Callable):
+    """
+    Generators are considered a special case of Callable that accept no
+    arguments and never memoize.
+    """
+
+    callable = param.Parameter(default=None, constant=True, doc="""
+         The generator function being wrapped.""")
+
+    @property
+    def argspec(self):
+        from inspect import ArgSpec
+        return ArgSpec(args=[], varargs=None, keywords=None, defaults=None)
+
+    def __call__(self):
+        return next(self.callable)
```

Comment:
The ``next`` built-in takes no arguments so as before, there is nothing to do.

Actually that isn't *quite* right. In the Python help:

```
next(...)
    next(iterator[, default])
```

Where ``default`` is output instead of raising StopIteration when raised. I don't think I've ever seen ``default`` used and always raising ``StopIteration`` on exhaustion is what I expect.